tensorflow_serving/servables/tensorflow/testdata/generate_saved_model_half_plus_two_mkl.py

```diff
@@ -0,0 +1,291 @@
+      signature_inputs, signature_outputs,
+      tf.saved_model.signature_constants.REGRESS_METHOD_NAME)
+
+
+# Possibly extend this to allow passing in 'classes', but for now this is
+# sufficient for testing purposes.
+def _build_classification_signature(input_tensor, scores_tensor):
+  """Helper function for building a classification SignatureDef."""
+  input_tensor_info = tf.saved_model.utils.build_tensor_info(input_tensor)
+  signature_inputs = {
+      tf.saved_model.signature_constants.CLASSIFY_INPUTS: input_tensor_info
+  }
+  output_tensor_info = tf.saved_model.utils.build_tensor_info(scores_tensor)
+  signature_outputs = {
+      tf.saved_model.signature_constants.CLASSIFY_OUTPUT_SCORES:
+          output_tensor_info
+  }
+  return tf.saved_model.signature_def_utils.build_signature_def(
+      signature_inputs, signature_outputs,
+      tf.saved_model.signature_constants.CLASSIFY_METHOD_NAME)
+
+
+def _generate_saved_model_for_half_plus_two(export_dir,
+                                            as_text=False,
+                                            use_main_op=False,
+                                            device_type="cpu"):
+  """Generates SavedModel for half plus two.
+
+  Args:
+    export_dir: The directory to which the SavedModel should be written.
+    as_text: Writes the SavedModel protocol buffer in text format to disk.
+    use_main_op: Whether to supply a main op during SavedModel build time.
+    device_name: Device to force ops to run on.
+  """
+  builder = tf.saved_model.builder.SavedModelBuilder(export_dir)
+
+  device_name = "/cpu:0"
+  if device_type == "gpu":
+    device_name = "/gpu:0"
+
+  with tf.Session(
+      graph=tf.Graph(),
+      config=tf.ConfigProto(log_device_placement=True)) as sess:
+    with tf.device(device_name):
+      # Set up the model parameters as variables to exercise variable loading
+      # functionality upon restore.
+      a = tf.Variable(0.5, name="a")
+      b = tf.Variable(2.0, name="b")
+      c = tf.Variable(3.0, name="c")
+
+      # Create a placeholder for serialized tensorflow.Example messages to be
+      # fed.
+      serialized_tf_example = tf.placeholder(tf.string, name="tf_example")
+
+      # Parse the tensorflow.Example looking for a feature named "x" with a
+      # single floating point value.
+      feature_configs = {
+          "x": tf.FixedLenFeature([1], dtype=tf.float32),
+          "x2": tf.FixedLenFeature([1], dtype=tf.float32, default_value=[0.0])
+      }
+      # parse_example only works on CPU
+      with tf.device("/cpu:0"):
+        tf_example = tf.parse_example(serialized_tf_example, feature_configs)
+      # Use tf.identity() to assign name
+
+      x = tf.identity(tf_example["x"], name="x")
+
+      # This operation just performs a convolution
+      # but the output is zero. This way we can keep
+      # the same output as the original code but trigger MKL-DNN.
+      o1 = tf.keras.layers.Conv2D(1,[1,1])(tf.zeros((1,10,10,1)))
```

Comment:
@netfs Thoughts on this approach?

Reply:
any reason why we cannot extend this:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/saved_model/saved_model_half_plus_two.py

to accommodate MKL, instead of creating a new copy here?
note our existing test saved_models are generated from this
script.

Reply:
@netfs the `half_plus_two` model is doing very simple operation, which is not invoking `mkl` ops. So @tonyreina added few convs to make sure `mkl` ops are used. He used above reference model file and updated with few convolutions. 

So do you have us to update above file and have some flags for mkl?

Reply:
and @netfs should we use bazel to generated saved models instead of python? Can you please direct @tonyreina .

Reply:
Hey Karthik. That looks good - after some internal discussion, we decided to move that file from tensorflow to tensorflow_serving since tensorflow_serving is the primary user of it.

Would you mind moving the edits to [this new file](https://github.com/tensorflow/serving/blob/master/tensorflow_serving/servables/tensorflow/testdata/saved_model_half_plus_two.py)? Apologies for the churn!

We can close the PR in tensorflow/tensorflow too. Thank you!

Reply:
yeah sure, will do it.